config/config.go

```diff
@@ -68,6 +68,73 @@ func NewCloudConfig(contents string) (*CloudConfig, error) {
 	return &cfg, err
 }
 
+func (co *CoreOS) Merge(mergeWith CoreOS) {
+	for _, u := range mergeWith.Units {
+		if unit := co.findUnit(u.Name); unit != nil {
+			unit.Merge(u)
+		} else {
+			co.Units = append(co.Units, u)
+		}
+	}
+}
+
+func (co *CoreOS) findUnit(unitName string) *Unit {
+	for i, u := range co.Units {
+		if u.Name == unitName {
+			return &co.Units[i]
+		}
+	}
+	return nil
+}
+
+func (cc *CloudConfig) Merge(mergeWith *CloudConfig) {
+	for _, key := range mergeWith.SSHAuthorizedKeys {
+		if !cc.sshKeyAlreadyExists(key) {
+			cc.SSHAuthorizedKeys = append(cc.SSHAuthorizedKeys, key)
+		}
+	}
+	for _, user := range mergeWith.Users {
+		if u := cc.findUser(user.Name); u != nil {
+			u.Merge(user)
+		} else {
+			cc.Users = append(cc.Users, user)
+		}
+	}
+	for _, file := range mergeWith.WriteFiles {
+		if !cc.writeFileAlreadyExists(file.Path) {
+			cc.WriteFiles = append(cc.WriteFiles, file)
+		}
+	}
```

Comment:
So there are other parts of the CloudConfig (hostname, manage_etc_host) -- why did you choose to merge some stuff but not others? How will that other stuff work if merged in?